qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -367,3 +376,270 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
         return common.generate_scheduling(
             instruction_durations, scheduling_method, timing_constraints, inst_map, target
         )
+
+
+class DefaultLayoutPassManager(PassManagerStagePlugin):
+    """Plugin class for default layout stage."""
+
+    def pass_manager(self, pass_manager_config, optimization_level=None) -> PassManager:
+        _given_layout = SetLayout(pass_manager_config.initial_layout)
+
+        def _choose_layout_condition(property_set):
+            return not property_set["layout"]
+
+        def _layout_not_perfect(property_set):
+            """Return ``True`` if the first attempt at layout has been checked and found to be
+            imperfect.  In this case, perfection means "does not require any swap routing"."""
+            return property_set["is_swap_mapped"] is not None and not property_set["is_swap_mapped"]
```

Comment:
Are there any concerns about pickle-ability of the resulting pass manager now we're retuning method-local functions as part of the conditions?

(also, lol at using explicitly private names when defining method-internal functions - a clear copy-paste sign haha)

Reply:
There shouldn't be as it's passing the parallel transpiler tests which should be testing this.